cpg_utils/hail_batch.py

```diff
@@ -101,7 +117,73 @@ def remote_tmpdir(hail_bucket: Optional[str] = None) -> str:
     return f'gs://{bucket}/batch-tmp'
 
 
-def dataset_path(suffix: str, category: Optional[str] = None) -> str:
+class PathScheme(ABC):
+    """
+    Cloud storage path scheme. Constructs full paths to buckets and files.
+    """
+
+    @abstractmethod
+    def path_prefix(self, dataset: str, category: str) -> str:
+        """Build path prefix used in dataset_path"""
+
+    @abstractmethod
+    def full_path(self, prefix: str, suffix: str) -> str:
+        """Build full path from prefix and suffix"""
+
+    @staticmethod
+    def parse(val: str) -> 'PathScheme':
+        """Parse subclass name from string"""
+        if val == 'gs':
+            return GSPathScheme()
+        if val in ['az', 'hail-az']:
+            return AzurePathScheme()
+        raise ValueError(f'Unsupported path format: {val}. Available: gs, hail-az')
+
+
+class GSPathScheme(PathScheme):
+    """
+    Google Cloud Storage path scheme.
+    """
+
+    def __init__(self):
+        self.scheme = 'gs'
+        self.prefix = 'cpg'
+
+    def path_prefix(self, dataset: str, category: str) -> str:
+        """Build path prefix used in dataset_path"""
+        return f'{self.prefix}-{dataset}-{category}'
+
+    def full_path(self, prefix: str, suffix: str) -> str:
+        """Build full path from prefix and suffix"""
+        return os.path.join(f'{self.scheme}://', prefix, suffix)
+
+
+class AzurePathScheme(PathScheme):
+    """
+    Azure Blob Storage path scheme, following the Hail Batch hail-az format.
+    """
+
+    def __init__(self, account: Optional[str] = 'cpg'):
+        config = get_config()
+        self.scheme = 'hail-az'
+        self.account = config['workflow'].get('azure_account', account)
```

Comment:
```suggestion
        self.scheme = 'hail-az'
        self.account = get_config()['workflow'].get('azure_account', account)
```

Reply:
Any reason why it's better?

Reply:
Just that `config` isn't reused anywhere, so it's just one less variable to keep track of :)